shuffle/cli/src/shared.rs

```diff
@@ -407,44 +413,61 @@ pub struct NetworksConfig {
 }
 
 impl NetworksConfig {
-    pub fn default() -> Self {
+    pub fn default() -> Result<Self> {
         let mut network_map = BTreeMap::new();
-        network_map.insert("localhost".to_string(), Network::default());
-        NetworksConfig {
+        network_map.insert("localhost".to_string(), Network::localhost()?);
+        network_map.insert("trove".to_string(), Network::trove_testnet()?);
+        Ok(NetworksConfig {
             networks: network_map,
-        }
+        })
     }
 
-    pub fn url_for(&self, network_name: &str) -> Result<Url> {
-        let specified_network = self.networks.get(network_name).ok_or_else(|| {
+    pub fn get_network(&self, network_name: &str) -> Result<Network> {
+        Network::from(self.networks.get(network_name).ok_or_else(|| {
             anyhow!("Please add specified network to the ~/.shuffle/Networks.json")
-        })?;
-        Ok(Url::from_str(
-            format!(
-                "{}:{}",
-                specified_network.base, specified_network.dev_api_port
-            )
-            .as_str(),
-        )?)
+        })?)
     }
 }
 
 #[derive(Debug, Serialize, Deserialize, PartialEq)]
 pub struct Network {
     name: String,
-    base: String,
-    json_rpc_port: u16,
-    dev_api_port: u16,
+    json_rpc_url: Url,
+    dev_api_url: Url,
+    faucet_url: Option<Url>,
 }
 
 impl Network {
-    pub fn default() -> Self {
-        Network {
-            name: String::from(LOCALHOST_NETWORK_NAME),
-            base: String::from(LOCALHOST_NETWORK_BASE),
-            json_rpc_port: 8080,
-            dev_api_port: 8081,
-        }
+    pub fn localhost() -> Result<Self> {
+        Ok(Network {
+            name: String::from(LOCALHOST_NAME),
+            json_rpc_url: Url::from_str("http://127.0.0.1:8080")?,
+            dev_api_url: Url::from_str("http://127.0.0.1:8081")?,
```

Comment:
As of https://github.com/diem/diem/pull/9692 the default port for the dev api is 8080 (same as json rpc).